src/test/test_runner.ts

```diff
@@ -29,11 +29,45 @@ module.exports = {
 
 		return new Promise(async (resolve, reject) => {
 			try {
-				const files = await glob("**/**.test.js", { cwd: testsRoot });
+				let testPatterns = ["**/**.test.js"];
+
+				// Apply test filters if provided
+				const testFilter = process.env.DART_CODE_TEST_FILTER;
+				if (testFilter) {
+					try {
+						const filters = JSON.parse(testFilter);
+						if (Array.isArray(filters) && filters.length > 0) {
+							// Create patterns for each filter
+							testPatterns = filters.map((filter) => `**/*${filter}*.test.js`);
```

Comment:
Can we also make it so that if the `.test.js` suffix is included (eg. because you pasted the whole path to `npm test`), that this still works? We should require that all tests end with `.test.js`, it should just be optional for the patterns.